### PR TITLE
[Validator] Fix `FileTest` to fit in 32bits architecture

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Following https://github.com/symfony/symfony/pull/52258#discussion_r1369819785

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
But we do want to test the behavior on x86 when the limit is above PHP_INT_MAX. That's removing it, isn't it?

**Comment by stof** [comment]
@nicolas-grekas We would need a separate test for that, which would test that behavior. But the behavior on 32bits cannot be that we return an integer bigger than PHP_INT_MAX

**Comment by nicolas-grekas** [comment]
Let's add this test case here then?

**Comment by alexandre-daubois** [comment]
Good idea, I added tests for the 32 bits architecture

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php [hunk 1]
```diff
@@ -31,6 +31,22 @@ public function testMaxSize($maxSize, $bytes, $binaryFormat)
         $this->assertTrue($file->__isset('maxSize'));
     }
 
+    /**
+     * @dataProvider provideValidSizesForIntegerOverflow
+     */
+    public function testMaxSizeWithIntegerOverflow($maxSize, $bytes, $binaryFormat)
+    {
+        if (32 !== 8 * \PHP_INT_SIZE) {
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]:
```suggestion
        if (4 !== \PHP_INT_SIZE) {
```

(we do this in other places too)

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php [hunk 2]
```diff
@@ -52,6 +68,22 @@ public function testMaxSizeCanBeSetAfterInitialization($maxSize, $bytes, $binary
         $this->assertSame($binaryFormat, $file->binaryFormat);
     }
 
+    /**
+     * @dataProvider provideValidSizesForIntegerOverflow
+     */
+    public function testMaxSizeCanBeSetAfterInitializationWithIntegerOverflow($maxSize, $bytes, $binaryFormat)
+    {
+        if (32 !== 8 * \PHP_INT_SIZE) {
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]:
```suggestion
        if (4 !== \PHP_INT_SIZE) {
```

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php [hunk 3]
```diff
@@ -101,11 +133,17 @@ public static function provideValidSizes()
...
-            ['4G', 4000000000, false],
+            ['2g', 2000000000, false],
+            ['2G', 2000000000, false],
         ];
     }
 
+    public static function provideValidSizesForIntegerOverflow(): \Generator
+    {
+        yield ['4g', '4000000000', false];
+        yield ['4G', '4000000000', false];
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Constraints/FileTest.php:176]: I think we should also have a test for 64 bit systems exclusively that ensure the value is an integer.

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php (hunk 3 above)
Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php:176]: Or maybe add a flag to the existing data provider which indicates if the provided fixture exceeds `PHP_INT_MAX` and cast to string on 32-bit systems?

Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php:176]: see #52263 for what I have in mind

Comment (alexandre-daubois) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php:176]: I think your way is cleaner and better!

**Comment by alexandre-daubois** [comment]
Closing in favor of https://github.com/symfony/symfony/pull/52263

**Pull request closed without merging**